exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1550 @@
+	if err != nil {
+		return err
+	}
+
+	wsRunningEndpoint, err := by.API.Endpoints.GetURL(exchange.WebsocketSpot)
+	if err != nil {
+		return err
+	}
+
+	err = by.Websocket.Setup(
+		&stream.WebsocketSetup{
+			ExchangeConfig:        exch,
+			DefaultURL:            bybitWSBaseURL + wsSpotPublicTopicV2,
+			RunningURL:            wsRunningEndpoint,
+			RunningURLAuth:        bybitWSBaseURL + wsSpotPrivate,
+			Connector:             by.WsConnect,
+			Subscriber:            by.Subscribe,
+			Unsubscriber:          by.Unsubscribe,
+			GenerateSubscriptions: by.GenerateDefaultSubscriptions,
+			Features:              &by.Features.Supports.WebsocketCapabilities,
+			OrderbookBufferConfig: buffer.Config{
+				SortBuffer:            true,
+				SortBufferByUpdateIDs: true,
+			},
+			TradeFeed: by.Features.Enabled.TradeFeed,
+		})
+	if err != nil {
+		return err
+	}
+
+	err = by.Websocket.SetupNewConnection(stream.ConnectionSetup{
+		URL:                  by.Websocket.GetWebsocketURL(),
+		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+	})
+	if err != nil {
+		return err
+	}
+
+	return by.Websocket.SetupNewConnection(stream.ConnectionSetup{
+		URL:                  bybitWSBaseURL + wsSpotPrivate,
+		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+		Authenticated:        true,
+	})
+}
+
+// AuthenticateWebsocket sends an authentication message to the websocket
+func (by *Bybit) AuthenticateWebsocket(ctx context.Context) error {
+	return by.WsAuth(ctx)
+}
+
+// Start starts the Bybit go routine
+func (by *Bybit) Start(wg *sync.WaitGroup) error {
+	if wg == nil {
+		return fmt.Errorf("%T %w", wg, common.ErrNilPointer)
+	}
+	wg.Add(1)
+	go func() {
+		by.Run()
+		wg.Done()
+	}()
+	return nil
+}
+
+// Run implements the Bybit wrapper
+func (by *Bybit) Run() {
+	if by.Verbose {
+		log.Debugf(log.ExchangeSys,
+			"%s Websocket: %s.",
+			by.Name,
+			common.IsEnabled(by.Websocket.IsEnabled()))
+		by.PrintEnabledPairs()
+	}
+
+	if !by.GetEnabledFeatures().AutoPairUpdates {
+		return
+	}
+
+	err := by.UpdateTradablePairs(context.TODO(), false)
+	if err != nil {
+		log.Errorf(log.ExchangeSys,
+			"%s failed to update tradable pairs. Err: %s",
+			by.Name,
+			err)
+	}
+}
+
+// FetchTradablePairs returns a list of the exchanges tradable pairs
+func (by *Bybit) FetchTradablePairs(ctx context.Context, a asset.Item) ([]string, error) {
+	if !by.SupportsAsset(a) {
+		return nil, fmt.Errorf("asset type of %s is not supported by %s", a, by.Name)
+	}
+
+	switch a {
+	case asset.Spot:
+		allPairs, err := by.GetAllSpotPairs(ctx)
+		if err != nil {
+			return nil, err
+		}
+		pairs := make([]string, len(allPairs))
+		for x := range allPairs {
+			pairs[x] = allPairs[x].BaseCurrency + currency.DashDelimiter + allPairs[x].QuoteCurrency
+		}
+		return pairs, nil
+	case asset.CoinMarginedFutures, asset.USDTMarginedFutures, asset.Futures:
```

Comment:
I'm curious, where is BTC-PERP? https://bybit-exchange.github.io/docs/usdc/perpetual/#t-querysymbolinfo

Reply:
I think that would be part of `USDCMarginedFutures` which will need to be implemented and also `USDCMarginedOptions`. I am curious as well.

Reply:
So when we started with adding support for By bit and during the development `USDCMarginedFutures` and `USDCMarginedOptions` were not mentioned in the API document as per my observation. But I can see them now. 
I am open to suggestions on how we proceed now. @shazbert @thrasher- @gloriousCode 
Do we first close the PR without these 2 assets or do we add support for them and then close the PR.

Looking forward to hearing from you guys.

Reply:
I am all for implementing them right now as much as you can in regards to REST implementation, especially the USDC margined futures. Defer websocket handing like the others.

Reply:
Am fine with adding USDC Rest margined futures support in this PR since it's likely a copy of the USDT margin based futures, however the options one could be more challenging since GCT has no concept of them currently. Can just fill out the REST related API functions without any support/linkage to the wrappers so people who use GCT as a library can benefit from direct calls to those endpoints and will be ready to go once GCT supports options trading.